common/achievement/create.ts

```diff
@@ -39,7 +47,12 @@ async function createAchievement(templateToCreate: Achievement_template, userId:
     }
 }
 
-async function createNextUserAchievement(templatesForGroup: Achievement_template[], nextStepIndex: number, userId: string, context: UserAchievementContext) {
+async function createNextUserAchievement<ID extends ActionID>(
+    templatesForGroup: Achievement_template[],
+    nextStepIndex: number,
+    userId: string,
+    context: SpecificNotificationContext<ID>
+) {
     const nextStepTemplate = templatesForGroup.find((template) => template.groupOrder === nextStepIndex);
 
```

Comment:
For validation reasons: we could check here whether the correct context for the template exists using the `templateFor` field, as we know here if `templateFor === Match` that a `matchId` / or `relation: "match/1"` must exist, the same applies to subcourse

Reply:
but we do pass in the context from the previous userAchievement. Isn't that what we want here? like:
{ matchId: match/1 } from previous userAchievement gets also written in this achievements context.